### ISSUE TITLE
CREDIT CARD FRAUDDETECTION

### ISSUE THREAD

**Issue by floki02** [floki02/Codesoft#1]
`# -*- coding: utf-8 -*-
"""Creadit_card_1.ipynb

Automatically generated by Colab.

Original file is located at
    https://colab.research.google.com/drive/1XuUzSAnz4bbz-YFT71Pg1cARvudDViSh
"""

import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

#loding the dataset to a Pandas DataFrame
c_c_data = pd.read_csv('/content/creditcard.csv')

c_c_data.head()

c_c_data.tail()

c_c_data.info()

#cheching the number of each coloum
c_c_data.isnull().sum()

c_c_data = c_c_data.dropna()

c_c_data['Class'].value_counts()

#seperating the data
c_c_data_legit = c_c_data[c_c_data.Class == 0]
c_c_data_fraud = c_c_data[c_c_data.Class == 1]

print(c_c_data_legit.shape)
print(c_c_data_fraud.shape)

#mesures of data
c_c_data_legit.Amount.describe()

c_c_data_fraud.Amount.describe()

#compare the values
c_c_data.groupby('Class').mean()

c_c_data_legit_sample = c_c_data_legit.sample(n=102)

new_dataset = pd.concat([c_c_data_legit_sample, c_c_data_fraud], axis=0)

new_dataset.head()

new_dataset.tail()

new_dataset['Class'].value_counts()

new_dataset.groupby('Class').mean()

from re import X
X = new_dataset.drop(columns='Class', axis=1)
Y = new_dataset['Class']

print(X)

print(Y)

X_train, X_text,Y_train, Y_test = train_test_split(X, Y, test_size=0.2, stratify=Y, random_state=2)

print(X.shape, X_train.shape, X_text.shape)

model = LogisticRegression()

model.fit(X_train, Y_train)

X_train_prediction = model.predict(X_train)
training_data_accuracy = accuracy_score(X_train_prediction, Y_train)

print('Accuracy on Training data : ', training_data_accuracy)

X_text_prediction = model.predict(X_text)
text_data_accuracy = accuracy_score(X_text_prediction, Y_test)

print('Accuracy score on Test Data : ', text_data_accuracy)

`
